### DISCUSSION TITLE
How to get a better level?

### DISCUSSION THREAD

**Discussion by armishmajerj** [bassamanator/everything-sovol-sv06#2] [General]
![image](https://user-images.githubusercontent.com/111510508/220520868-47731332-df3b-4cb4-b35b-3d54b16e3fc7.png)
Above is my current bed level mesh from Klipper. I have tried numerous things to get this more leveled. I have re-greased the bearings and rods. I have tried lowering the gantry almost to bed and using two identically sized objects to manually adjust the screws on the bottom holding the z screws in place. This works for a while but it seems like the z screw particularly on the right side gets out of wack over time.

I'm fairly new to 3d printing in general but have changed over to using your profile after setting it up originally. I do have another question - why does your profile seem to produce a high frequency whine over the stock? I would imagine it has something to do with the microsteps number in the stepper section of the printer.cfg

**Comment by armishmajerj** [comment]
Here are two videos of what I see happening while creating the bed mesh and printing in general. The right side z screw seems to wobble? I tried checking it was straight when I lubed the bearings/rods and it seemed to be.

https://user-images.githubusercontent.com/111510508/220617697-3eab8912-af87-4b72-8808-c8721c5b143a.mov
^Left Side
https://user-images.githubusercontent.com/111510508/220617706-e9e7832a-584e-43e2-bb0a-49a3a1457adf.mov
^Right Side

**Comment by bassamanator** [comment]
No lead screws are perfect. If you want perfection, you have to buy $1000 ball screws, those are perfect and mostly unbendable. You can buy a pair of lead screws that are used in this printer for about $10. Lead screws have one task, to lift the gantry, that's it. The lead rods have the task of keeping things aligned properly. So, long story short, your lead screws are fine. I would *highly* recommend not putting bearings at the top of the lead screws, they have the potential to create more problems than they solve.

Please play with the `microsteps` setting to reduce the whine: 16,32,64 should all be valid options. You probably won't even see a difference. One thing to try first would be to enable stealthchop by adding `stealthchop_threshold: 999999` under all the `[tmc2209 stepper_]`. You can also try `interpolate: True`. Don't be afraid to play with these settings, they will not harm your printer, and you can always revert back to my config (or someone elses). [Ref](https://www.klipper3d.org/TMC_Drivers.html?h=stealthchop_threshold#setting-spreadcycle-vs-stealthchop-mode). 

Now on to the problem at hand, the not-so-level-bed. That is a major problem with this printer, I am suffering myself. (Can you also please a picture with 0, 0 at the bottom left, and more of an overhead shot?). If you had one point that was very low on the mesh, and the reset were higher, the [Relative Reference Index](https://www.klipper3d.org/Bed_Mesh.html?h=relative#the-relative-reference-index) would help, but based on what I'm seeing, I don't think it will. 

These mods do help a bit, I did the silicone version myself:
- https://www.rearvuemirror.com/guides/nylock-mod-for-the-mk3s probably easiest
- https://github.com/bbbenji/PMSBLM 

My solution has been to get the bed and nozzle hot (60/180) with no filament. After I have already adjusted z_offset via `probe_calibrate`, I go to the 9 obvious points on the bed, and bring down the nozzle gradually (sometimes even going up) to 0.0 or 0.1 and I do the paper test. If nozzle is just touching the paper at the center of the bed with the nozzle at z = 0.1, but it's touching the bed at x = 0 at the same z height, I remove the PEI sheet and lower the bed on that side by turning the bed-screws clockwise. I do this everywhere, making sure that at z = 0.0, the nozzle will not `dig` into the PEI sheet anywhere. Nothing will ever be perfect with this method because there are only 5 screws on this bed, however, you can get to a point that the nozzle will not damage the bed, and the mesh will take care of most things. 

Hope this helps and keep this discussion going.

PS You can also contact Sovol about this. The issue is that the first layer is so essential to a good, accurate print, and with the `hacks` that I've mentioned above, you still will not have a perfect first layer over the entirety of your bed.

EDIT use this part as a reference point for yourself will doing the bed adjustments https://www.printables.com/model/43629-prusa-mini-silicone-bed-leveling-mod-bed-tightenin

**Reply by armishmajerj** [reply]
Here is a picture of the mesh from the angle I think you are requesting - edited to get right one for you
![image](https://user-images.githubusercontent.com/111510508/220806419-94a99112-0e1e-46b5-b341-9f351afe0d9a.png)

I'm not 100% on what you mean by the 9 obvious points on the bed? Do you mean the 5 screw locations and in between? Do you just send the command to move the printer to the relevant X,Y location for each point and then set Z = 0? Just ensure everywhere basically on bed Z = 0 doesn't scratch it and let bed mesh handle rest? I've printed quite a few things already since getting this machine :D just run into issues it seems where the right side z screw comes uneven after some time and I don't get why that is happening.

For the nylock mod I assume just do it to the 5 screws on the bed? The guide mentions 9 but that seems for the Prusa mk3s. Figure I should make it extra clear in case anyone else in the future is curious.

My worry on contacting Sovol is that I've seen mention that flashing other firmware on the board voids any kind of warranty but also from what I understand I could just flash it back before sending it back if needed.

For reference this is my history since klipperizing this device the day after I got it - had plenty of prints come out really great! Just have had some longer prints fail in the final hour and trying to narrow down why can be frustrating! Especially when everyone on reddit just wants to blame Sovol when yes there are some things they didn't get right but for my first 3d printer and the number of cool things I've printed I love it!
![image](https://user-images.githubusercontent.com/111510508/220803221-d63cf3c6-db3b-4578-8b92-6bea46bea287.png)

**Comment by bassamanator** [comment]
You can flash marlin back, the same way you flashed Klipper. I've done it many times. This is not a firmware problem, this is a hardware problem, straight from the manufacturer, extremely common for this printer actually. You need not mention anything about the firmware btw.

For the nylock mod, you need 4, keep the center spacer in place as a reference. This video is somewhat [useful](https://www.youtube.com/watch?v=qRbMOfMy-MA&t=1094s).

Roughly speaking, think of tic-tac-toe for the 9 points. Check 0, 0, bring it down to z 0.2, then 0.1, then 0.0 (sometimes when you're very close to the bed, and try to go to 0, it will actually move the nozzle up, this is normal, so go back up a bit, then bring it down). At z 0.0, the nozzle should just be exactly touching the bed, not digging into it. If that is the case, then check 223, 0, and so on and so forth, while adjusting the bed screws as needed. Things will not be perfect, you just want to avoid digging.

Impressive print time btw. And failed prints happen to the most experienced printers, it's normal. And don't be afraid to disassemble the lead screws to inspect them closely. Bring the gantry almost all the way to the top, remove the top plastic holder parts, remove the gantry, unscrew the lead screws and see if there is a serious deviation. Something that I saw in a `Teaching Tech` video, was putting a small ball bearing between the motor shaft and the lead screw, see [this](https://youtu.be/snM_NXr4SCo?t=345), it might be relevant.

And keep this thread going! 😁

**Reply by armishmajerj** [reply]
Interesting so for the video from Teaching Tech would I be able to use that same stl file from this https://www.printables.com/model/88587-z-axis-stepper-motor-mount-with-integrated-axial-t ? Or after looking around I think this may be for the sv06 while accomplishing the same thing https://www.printables.com/model/364758-z-axis-bracket-upgrade-sv06/comments ? I don't have PETG to print in and have yet to try. If this is something that would help I would happily go pick up a spool from Microcenter and try printing these out! I just had the same print fail around the same spot in a 12 hour print despite my bed mesh looking much better https://cults3d.com/en/3d-model/various/infernal-dragon-hell-heaven - this is what I am trying to print! Looks really cool and I've done some other articulated designs from this creator that worked out well but just having a hard time on this one but I will print some other things until I can get this issue sorted out

**Comment by bassamanator** [comment]
Actually, I wasn't recommending that you do this mod, only that **there might be a benefit from making sure the lead screw and the Z stepper motor are *not* in contact with each other**. If you try this, just make sure the ball bearing is small enough so that it doesn't touch the plastic housing at the top.

**Comment by bassamanator** [comment]
>why does your profile seem to produce a high frequency whine over the stock? 

Just wanted to mention that increasing the microsteps actually reduces the 'whining' sound. I have set x, y, z `microsteps: 64`, and it's whisper quiet. Will adjust the `printer.cfg` after my 3-hour-long test print completes.

**Reply by transistor1** [reply]
@armishmajerj  is it possible the drop-off to the right side is the inductive sensor extending over the side of the bed?  I noticed it was doing that on mine, and shortened the dimensions of the bed in Klipper.

I am also having a hard time getting a good level across the entire bed as well.  I actually did the tubing mod on my SV06 that @bassamanator  mentioned, and it did help get a much flatter surface, but it still does not produce a uniform bottom layer across the whole bed.

Here's my klipper bed mesh, below.  My previous range before the mod was about 0.9
<img width="1013" alt="Screen Shot 2023-03-06 at 9 41 52 PM" src="https://user-images.githubusercontent.com/5625360/223307441-c88d2892-8ac0-45ca-99d2-d78e5d06d4e3.png">

**Comment by bassamanator** [comment]
@transistor1 The sensor remains on top of the bed at all times, have a look next time you create a mesh.

The bed flatness is a serious issue on this printer. I will soon attach a CRtouch just to see if the inductive sensor is behaving badly for some reason.

**Comment by transistor1** [comment]
> @transistor1 The sensor remains on top of the bed at all times, have a look next time you create a mesh.

Mine does now, but it didn't when I was first configuring Klipper, maybe a month or so ago. I had to shrink the max bed values in bed_mesh to prevent that. Your config probably contains accurate bed values. At the time I think I was trying to config from a YouTube video and probably missed something. It's worth checking into. 

It looked like this while it was creating the mesh:

![image](https://user-images.githubusercontent.com/5625360/223852084-723be841-755b-45be-81f2-4230fd1f5843.jpeg)

**Reply by bassamanator** [reply]
Very curious. Was this with my OSS klipper config?

**Comment by transistor1** [comment]
No, it wasn't. I was configuring from a YouTube video (I edited my comment
to reflect that after I replied to you). This was a while ago, too.  Just
meant this as a tip to have the OP check if it's happening for them, in
case they inadvertently changed something.

On Wed, Mar 8, 2023 at 4:29 PM Bassam Husain ***@***.***>
wrote:

> Very curious. Was this with my OSS klipper config?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bassamanator/everything-sovol-sv06/discussions/2#discussioncomment-5246593>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK5MECVKJBHYVIVF5YJW23W3D22VANCNFSM6AAAAAAVD2RINE>
> .
> You are receiving this because you were mentioned.Message ID:
> <bassamanator/everything-sovol-sv06/repo-discussions/2/comments/5246593@
> github.com>
>
-- 
Sent from my phone

**Reply by armishmajerj** [reply]
I'm away from my house at the moment but I think you've got it 😂 was this nero3d? I know when I started using bassamanator config I didn't copy all settings over. This is likely the reason I will check and confirm when I get home. Can one of you point me where this is? I'm sure it's obvious and I'll find it right away

**Reply by bassamanator** [reply]
@armishmajerj Here you go: https://github.com/bassamanator/Sovol-SV06-firmware

**Reply by transistor1** [reply]
@armishmajerj specific section of printer.cfg:

```ini
[bed_mesh]
speed: 175
mesh_min: 26.4, 31
mesh_max: 210, 201
probe_count: 15, 15
horizontal_move_z: 5
algorithm: bicubic
fade_start: 1
fade_end: 10
fade_target: 0
```

**Reply by transistor1** [reply]
@armishmajerj BTW I would use whatever values @bassamanator is using for that b/c my printer is clearly not optimal rn 😂

**Reply by transistor1** [reply]
```ini
mesh_min: 27, 5
mesh_max: 222, 203
```

**Reply by armishmajerj** [reply]
Darn, that's what mine actually is. Well I will figure this out eventually. I did the silicone mod and that has helped get me back to printing but I am hoping to have more time this weekend to try fixing it.

**Comment by transistor1** [comment]
Yes it was Nero3d!  He did a live stream on the sv06.

On Wed, Mar 8, 2023 at 4:51 PM Jerj ***@***.***> wrote:

> I'm away from my house at the moment but I think you've got it 😂 was
> this nero3d? I know when I started using bassamanator config I didn't copy
> all settings over. This is likely the reason I will check and confirm when
> I get home. Can one of you point me where this is? I'm sure it's obvious
> and I'll find it right away
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bassamanator/everything-sovol-sv06/discussions/2#discussioncomment-5246765>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK5MEBJXNKVVBWJ2E2Y6J3W3D5MPANCNFSM6AAAAAAVD2RINE>
> .
> You are receiving this because you were mentioned.Message ID:
> <bassamanator/everything-sovol-sv06/repo-discussions/2/comments/5246765@
> github.com>
>
-- 
Sent from my phone

**Comment by armishmajerj** [comment]
I did end up installing the silicone mod on both my printers and that helped a ton to reduce the overall variation to < .2 mm which is an improvement over the original. I also set up KAMP https://github.com/kyleisah/Klipper-Adaptive-Meshing-Purging which has allowed me to only create a bed mesh for the area of the bed I am printing 
![image](https://user-images.githubusercontent.com/111510508/233073030-a4bc75fd-c7e7-4ec7-af35-cc31f09dcc1b.png)
